pkg/application/client.go

```diff
@@ -63,10 +67,381 @@ func (c *ApplicationClient) List(ctx context.Context, limit, offset int) (*Appli
 	}
 	resp, err := c.apiClient.GetAllApplicationsWithResponse(ctx, &params)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list applications: %w", err)
+		return nil, fmt.Errorf("Failed to list applications: %w", err)
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("Failed to list applications: status %d, body: %s", resp.StatusCode(), string(resp.Body))
+	}
+	return resp.JSON200, nil
+}
+
+// CreateSinglePageApp creates a new Single Page Application with sensible defaults
+func (c *ApplicationClient) CreateSinglePageApp(ctx context.Context, name string, redirectURL string) (*ApplicationBaseModel, error) {
```

Comment:
IMO, we don't need these functions in the SDK level as using the createApplication function with proper templateId would do the job. That's the approach the competitor SDKs follow too. maybe we can improve the createApplication function to pass a template type and internally map it to the UUID when building the request.

Reply:
These functions will be very useful for the mcp server though. it'd be better to move these to mcp server side as tools.

Reply:
My view is these are needed. We can have a generic function as well. But specific functions improve the usability.

Reply:
The purpose of an SDK is to improve experience for the developer as much as possible. Having everything generic again push the developer to handle lot of stuff their own

Reply:
> The purpose of an SDK is to improve experience for the developer as much as possible. Having everything generic again push the developer to handle lot of stuff their own

Yes, agreed. My concern was the exact mapping of functions to UI templates. Whenever we do a modification to UI templates, we will have to update the SDK methods as well. 

Maybe we can deliver the same DX improvement by letting user pass the application type as a parameter to createApplication function, as @shashimalcse also has mentioned above. WDYT?

Reply:
Well this UX is overall designed to get the app to a working state with minimal configs. So I don't quite think new stuff will onboard there. We can have the discussed generic app creation function to facilitate such then, or have more overloaded methods based on the requirement